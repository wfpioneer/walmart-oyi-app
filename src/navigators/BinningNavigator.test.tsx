import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {
  BinningNavigatorStack, renderKebabButton, renderScanButton, resetManualScan
} from './BinningNavigator';

describe('Binning Navigator', () => {
  it('Renders the Binning navigator component', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BinningNavigatorStack
        isManualScanEnabled={false}
        dispatch={jest.fn}
        managePalletMenu={false}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the scanButton header icon for Binning Screen when rightmost button', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      renderScanButton(jest.fn(), false, true)
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the scanButton header icon for Binning Screen when not rightmost button', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      renderScanButton(jest.fn(), false, false)
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders the kebab button for the binning screen', () => {
    const renderer = ShallowRenderer.createRenderer();

    renderer.render(
      renderKebabButton(jest.fn(), jest.fn())
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Expects dispatch to be called if isManualScanEnabled is true for "resetManualScan()"', () => {
    const mockDispatch = jest.fn();
    const manualScanEnabled = true;
    resetManualScan(manualScanEnabled, mockDispatch);
    expect(mockDispatch).toHaveBeenCalled();
  });
});
