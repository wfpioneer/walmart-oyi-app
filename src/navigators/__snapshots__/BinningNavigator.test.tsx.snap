// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Binning Navigator Renders the Binning navigator component 1`] = `
<StackNavigator
  screenOptions={
    Object {
      "headerMode": "float",
      "headerStyle": Object {
        "backgroundColor": "#0071e9",
      },
      "headerTintColor": "#ffffffff",
    }
  }
>
  <Screen
    component={[Function]}
    listeners={
      Object {
        "beforeRemove": [Function],
        "blur": [Function],
      }
    }
    name="BinningHome"
    options={
      Object {
        "headerRight": [Function],
        "headerTitle": "[missing \\"en.BINNING.BINNING\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="AssignLocation"
    options={
      Object {
        "headerRight": [Function],
        "headerTitle": "[missing \\"en.BINNING.ASSIGN_LOCATION\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="ManagePallet"
    options={
      Object {
        "headerRight": [Function],
        "headerTitle": "[missing \\"en.PALLET.MANAGE_PALLET\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="CombinePallets"
    options={
      Object {
        "headerRight": [Function],
        "headerTitle": "[missing \\"en.PALLET.COMBINE_PALLETS\\" translation]",
      }
    }
  />
</StackNavigator>
`;

exports[`Binning Navigator Renders the scanButton header icon for Binning Screen when not rightmost button 1`] = `
<TouchableOpacity
  hostRef={null}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 15,
      }
    }
  >
    <mockConstructor
      color="#ffffffff"
      name="barcode-scan"
      size={20}
    />
  </View>
</TouchableOpacity>
`;

exports[`Binning Navigator Renders the scanButton header icon for Binning Screen when rightmost button 1`] = `
<TouchableOpacity
  hostRef={null}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 15,
      }
    }
  >
    <mockConstructor
      color="#ffffffff"
      name="barcode-scan"
      size={20}
    />
  </View>
</TouchableOpacity>
`;

exports[`Binning Navigator renders the kebab button for the binning screen 1`] = `
<TouchableOpacity
  hostRef={null}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 15,
      }
    }
  >
    <mockConstructor
      color="#ffffffff"
      name="dots-vertical"
      size={30}
    />
  </View>
</TouchableOpacity>
`;
